lib/scenic/adapters/postgres.rb

```diff
@@ -155,17 +157,34 @@ def create_materialized_view(name, sql_definition, no_data: false)
       # @param no_data [Boolean] Default: false. Set to true to create
       #   materialized view without running the associated query. You will need
       #   to perform a non-concurrent refresh to populate with data.
+      # @param side_by_side [Boolean] Default: false. Set to true to create the
+      #   new version under a different name and atomically swap them, limiting
+      #   the time that a view is inaccessible at the cost of doubling disk usage
       #
       # @raise [MaterializedViewsNotSupportedError] if the version of Postgres
       #   in use does not support materialized views.
       #
       # @return [void]
-      def update_materialized_view(name, sql_definition, no_data: false)
+      def update_materialized_view(name, sql_definition, no_data: false, side_by_side: false)
         raise_unless_materialized_views_supported
 
-        IndexReapplication.new(connection: connection).on(name) do
-          drop_materialized_view(name)
-          create_materialized_view(name, sql_definition, no_data: no_data)
+        if side_by_side
+          session_id = Time.now.to_i
+          new_name = generate_name name, "new_#{session_id}"
+          drop_name = generate_name name, "drop_#{session_id}"
+          IndexReapplication.new(connection: connection).on_side_by_side(
+            name, new_name, session_id
+          ) do
+            create_materialized_view(new_name, sql_definition, no_data: no_data)
+          end
+          rename_materialized_view(name, drop_name)
+          rename_materialized_view(new_name, name)
+          drop_materialized_view(drop_name)
```

Comment:
Why do we:

1. rename existing view
2. rename new view
3. drop existing view

instead of:

1. drop existing view
2. rename new view

In the first case, the view is no longer available to the application anyway. Is it that `drop` is potentially slow?

Reply:
Not sure. OP may know; I was just rearranging

Reply:
To guarantee that if you're outside a transaction, there's no potential outage window where the view is missing.